File: ABC177/C_Sum_of_product_of_pairs/submission1.java
// TLE

import java.util.Scanner;

class Main {
  public static void main(String[] args) {

    var m = 1000000000 + 7;

    var sc = new Scanner(System.in);

    var n = sc.nextInt();
    var a = new int[n];
    for (int i = 0; i < n; i++) {
      a[i] = sc.nextInt() % m;
    }

    long answer = 0;
    for (int i = 0; i < n; i++) {
      for (int j = i + 1; j < n; j++) {
        answer = (answer + (long)a[i] * a[j] % m) % m;
      }
    }

    System.out.println(answer);

  }
}

File: ABC097/D_Equals/submission1.scala
// TLE

import java.util.Scanner;

object Main extends App {
  val sc = new Scanner(System.in);

  val n, m = sc.nextInt();

  val ps = List.fill(n)(sc.nextInt() - 1);

  val xy = List.fill(m)((sc.nextInt() - 1, sc.nextInt() - 1));

  val root = (0 until n).toArray;

  def fetchRoot(i: Int): Int = {
    val r = root(i);
    if (r == i) {
      r;
    } else {
      val r2 = fetchRoot(r);
      root(i) = r2;
      r2;
    }
  }

  xy.foreach { case (x, y) =>
    val rx = fetchRoot(x);
    val ry = fetchRoot(y);
    if (rx != ry) {
      root(ry) = rx;
    }
  }

  val answer = (0 until n).count { i =>
    val p = ps(i);
    val rp = fetchRoot(p);
    val ri = fetchRoot(i);
    ri == rp;
  }

  println(answer);
}
